environment_tools/node_utils.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+import re
+
+from environment_tools import type_utils
+
+NODE_NAME_FORMAT = '[A-Za-z0-9]*(-[A-Za-z0-9]*)*-(?P<location>[A-Za-z0-9]*)($|\..*)'
+
+
+def get_node_location(node, name_format=NODE_NAME_FORMAT):
+    """ Parses the node location from the provided node hostname.
+
+    :param node: A string that reprsents the node's hostname eg "my-machine-devb.place.com"
+    :param name_format: A string that represents the regex to parse the node's location from
+        the full hostname. Must have the capture group 'location' defined to correctly parse
+        the location.
+    :returns: node_location, The location of the given hostname or None if not found.
+    :rtype: String or None
+    """
+    matches = re.search(name_format, node)
+    if not matches:
+        return None
+    else:
+        node_location = matches.group('location')
+        return node_location
+
+
+def get_node_location_type(node_location):
```

Comment:
This is pretty iffy. What would 'devc' return?

I think this function is not something we can do, and by making it we have a high chance of introducing bugs where we are making assumptions that there is "only one" node location for a string

Reply:
Ah good point. Do you think returning a list of valid node locations would be a decent alternative?

Reply:
I'm not sure this function needs to exist. Hostnames in our environment should always end in the habitat, the most specific location_type, so there's no need to figure out what location_type it is.